### ISSUE TITLE
Qemu modified binary images folder path is not used with old topologies

### ISSUE THREAD

**Issue by jean-christophe-manciot** [GNS3/gns3-gui#1016]
I modified the binary images folder path.
In the topology file, there is no mention of the old or new path, just the image name, such as:
"hda_disk_image": "vios-adventerprisek9-m.vmdk", or  "hda_disk_image": "ASAv-9.5.1.qcow2",.

However, when starting the node, I get the following error:
qemu-system-x86_64: -drive file=/home/actionmystique/Program-Files/Labs/GNS3/CCNP/TBSHOOT/L3-Technologies/IPv4/VRF/Jean-Christophe_Manciot/1+/Leaf & Spine DC-3-solutions-linux/project-files/qemu/c67ee2b6-1bd5-4dc0-904a-15538c079bd5/hda_disk.qcow2,if=ide,index=0,media=disk: Could not open backing file: **Could not open '/media/actionmystique/SAMSUNG-Ext4/Images/GNS3/QEMU/vios-adventerprisek9-m.vmdk': No such file or directory**

It is using the old images folder, not the new one.

When checking the GNS3 config files **in ~/.config/GNS3, in gns3_gui.conf or gns3_server.conf, there is no trace of the old path, only the new one.**
How is that issue even possible? Where could there be a backup of the old binary path?

**Comment by noplay** [comment]
It's by design in qemu. The qcow2 keep a reference to the old file in order to not duplicate the data. You can patch with qemu-img in order to relocate the data.

@grossmj perhaps we can check in the qcow2 at each startup if the file need to relocate. It doesn't seem a complicate to read this information:
https://people.gnome.org/~markmc/qcow-image-format.html

And after that if we need to update the file location we can run qemu-img.

**Comment by jean-christophe-manciot** [comment]
You mean that no one else can read the same lab, unless they have the exact same binary images folder path?

**Comment by jean-christophe-manciot** [comment]
Well, what you suggested works; changing the "backing file" from the old path to the new one is very simple:
# **qemu-img info hda_disk.qcow2** 
image: hda_disk.qcow2
file format: qcow2
virtual size: 2.2G (2361393152 bytes)
disk size: 17M
cluster_size: 65536
backing file: /media/actionmystique/SAMSUNG-Ext4/Images/GNS3/QEMU/Ubuntu-Server-15.04.qcow2
Format specific information:
    compat: 1.1
    lazy refcounts: false
    refcount bits: 16
    corrupt: false
# **qemu-img rebase -b /media/actionmystique/SAMSUNG-Ext4-1/Images/GNS3/QEMU/Ubuntu-Server-15.04.qcow2 hda_disk.qcow2 **
# qemu-img info hda_disk.qcow2 
image: hda_disk.qcow2
file format: qcow2
virtual size: 2.2G (2361393152 bytes)
disk size: 17M
cluster_size: 65536
backing file: /media/actionmystique/SAMSUNG-Ext4-1/Images/GNS3/QEMU/Ubuntu-Server-15.04.qcow2
Format specific information:
    compat: 1.1
    lazy refcounts: false
    refcount bits: 16
    corrupt: false

Basically, **qemu-img rebase -b new-image-path hdx_disk.qcow2** for all HDDs of all qemu nodes for the current lab will do the trick. 
However, it is very tedious and I don't expect the people who will receive the labs to go down that road.

**Comment by noplay** [comment]
This issue was moved to GNS3/gns3-server#466
